### PR TITLE
LibGfx: Convert strokes to fills => Nice stroke rendering for free!

### PR DESCRIPTION
**Pull request by MacDue**
Finally, I can be happy with the SVG rasterization! Note: There's still a bunch of TODOs (e.g. implement line joins other than round, wire up using paint styles with strokes, and probably fix bugs and optimise things).

| Old                                                                                                                                 | New                                                                                                                                 |
|-------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------|
| image                               | image                               |
| Screenshot from 2023-06-05 21-54-05 | Screenshot from 2023-06-05 21-51-21 |
| Screenshot from 2023-06-05 21-54-39 | Screenshot from 2023-06-05 21-51-35 |
| Screenshot from 2023-06-05 21-55-13 | Screenshot from 2023-06-05 21-52-17 |
| Screenshot from 2023-06-05 21-55-58 | image                               |

### CODE REVIEW

File: Userland/Libraries/LibGfx/Path.cpp [hunk 1]
```diff
@@ -365,4 +365,166 @@ void Path::add_path(Path const& other)
...
+    struct ActiveRange {
+        float start;
+        float end;
+
+        bool in_range(float angle) const
+        {
+            // Note: Since active ranges go counterclockwise start > end unless we wrap around at 180 degrees
+            return ((angle <= start && angle >= end)
+                || (start <= end && angle <= start)
+                || (start <= end && angle >= end));
```
Comment (GMTA) [inline Userland/Libraries/LibGfx/Path.cpp:456]: These last two lines probably don't hold when `start == end`, check should maybe be `start < end` instead?

File: Userland/Libraries/LibGfx/Path.cpp [hunk 2]
```diff
@@ -365,4 +365,166 @@ void Path::add_path(Path const& other)
...
+            return angle_between(shape[shape_idx], shape[shape_idx + 1]);
+        };
+
+        auto start_slope = slope();
+        auto active = 0u;
+        for (; active < active_ranges.size(); active++) {
+            if (active_ranges[active].in_range(start_slope)) {
+                break;
+            }
+        }
```
Comment (GMTA) [inline Userland/Libraries/LibGfx/Path.cpp:503]: If all you need is the index, maybe `Vector::find_first_index_if` can help here?

File: Userland/Libraries/LibGfx/Path.cpp [hunk 3]
```diff
@@ -365,4 +365,166 @@ void Path::add_path(Path const& other)
...
+
+        while (shape_idx < shape.size()) {
+            add_vertex(shape[shape_idx] + pen_vertices[active]);
+            auto slope_now = slope();
+            auto range = active_ranges[active];
+            if (range.in_range(slope_now)) {
+                shape_idx++;
+            } else {
+                if (clockwise(slope_now, range.end)) {
+                    if (active == unsigned(pen_vertex_count - 1))
```
Comment (GMTA) [inline Userland/Libraries/LibGfx/Path.cpp:513]: `static_cast<size_t>` instead of `unsigned(...)`?

**Review by GMTA — COMMENTED** [review]
Awesome stuff! Just some minor remarks.

File: Userland/Libraries/LibGfx/Path.cpp (hunk 1 above)
Comment (MacDue) [reply Userland/Libraries/LibGfx/Path.cpp]: I think so, I don't think there should be any case where `start == end` (for a circular pen)

File: Userland/Libraries/LibGfx/Path.cpp (hunk 2 above)
Comment (MacDue) [reply Userland/Libraries/LibGfx/Path.cpp]: Thanks! That's handy :)

**Review by GMTA — APPROVED** [review]

**Pull request merged**